<launch>
    <!-- ObjectDetection -->
    <arg name="SOURCE" default="/zed2/zed_node/rgb/image_rect_color" /> 
    <arg name="DEPTH_INPUT" default="/zed2/zed_node/depth/depth_registered" />
    <arg name="CAMERA_INFO" default="/zed2/zed_node/depth/camera_info" />
    <arg name="ROS_INPUT" default="True" />
    <arg name="USE_ACTIVE_FLAG" default="False" />
    <arg name="VERBOSE" default="False" />
    <arg name="DEPTH_ACTIVE" default="True" />
    <arg name="YOLO_BAG_MODEL_PATH" default="$(find bag_detector)/models/bolsasV1.pt" />
    <arg name="CAMERA_FRAME" default="zed2_left_camera_optical_frame" />
    <arg name="FLIP_IMAGE" default="False" />
    
    <node name="BagDetector" pkg="bag_detector" type="bagDetector.py" respawn="true" output="screen">
        <param name="SOURCE" value="$(arg SOURCE)" />
        <param name="ROS_INPUT" value="$(arg ROS_INPUT)" />
        <param name="USE_ACTIVE_FLAG" value="$(arg USE_ACTIVE_FLAG)" />
        <param name="VERBOSE" value="$(arg VERBOSE)" />
        <param name="YOLO_BAG_MODEL_PATH" value="$(arg YOLO_BAG_MODEL_PATH)" />
        <param name="DEPTH_ACTIVE" value="$(arg DEPTH_ACTIVE)" />
        <param name="DEPTH_INPUT" value="$(arg DEPTH_INPUT)" />
        <param name="CAMERA_INFO" value="$(arg CAMERA_INFO)" />
        <param name="CAMERA_FRAME" value="$(arg CAMERA_FRAME)" />
    </node>

    <node name="Pointing_Detector" pkg="bag_detector" type="detectPointingObject.py" respawn="true" output="screen">
        <param name="SOURCE" value="$(arg SOURCE)" />
        <param name="ROS_INPUT" value="$(arg ROS_INPUT)" />
        <param name="USE_ACTIVE_FLAG" value="$(arg USE_ACTIVE_FLAG)" />
        <param name="VERBOSE" value="$(arg VERBOSE)" />
        <param name="YOLO_BAG_MODEL_PATH" value="$(arg YOLO_BAG_MODEL_PATH)" />
        <param name="DEPTH_ACTIVE" value="$(arg DEPTH_ACTIVE)" />
        <param name="DEPTH_INPUT" value="$(arg DEPTH_INPUT)" />
        <param name="CAMERA_INFO" value="$(arg CAMERA_INFO)" />
        <param name="CAMERA_FRAME" value="$(arg CAMERA_FRAME)" />
    </node>

     <node name="person_tracking" pkg="vision" type="PersonTracking.py" respawn="true" output="screen">
        <param name="FLIP_IMAGE" value="$(arg FLIP_IMAGE)" />
    </node>
</launch>

